lib/ood_core/job/adapters/pbspro.rb

```diff
@@ -434,6 +453,21 @@ def parse_job_info(v)
               native: v
             )
           end
+
+          # Combine the array parent with the states of its children
+          def process_job_array(jobs)
+            parent = jobs.select { |j| /\[\]/ =~ j.id }.first.to_h
```

Comment:
I guess its a fair assumption, but this code assumes the array of jobs has at least one `Info` object with the id in the form of "parent[]". Crash occurs if thats not the case (if first returns nil)

Reply:
This method is only called once and is protected by an if/elsif: https://github.com/OSC/ood_core/blob/pbs-array/lib/ood_core/job/adapters/pbspro.rb#L336-L342